### DISCUSSION TITLE
Skypilot reuse hard drives

### DISCUSSION THREAD

**Discussion by yuval-airis** [skypilot-org/skypilot#2559] [Q&A]
I plan to use Skypilot on (Mainly AWS) preemptible instances ([Spots](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-spot-instances.html)). My code requires a lot of input coming through the hard drive, which I would rather not recreate on every machine creation.

Is it possible to request Skypilot to reuse given (possibly read-only) hard drives between runs/on preemptions?

**Comment by concretevitamin** [comment]
Is it possible to use a S3 bucket? It can be mounted to each SkyPilot cluster.  It is natively supported in Managed Spot (https://skypilot.readthedocs.io/en/latest/examples/spot-jobs.html#checkpointing-and-recovery) either for read-only datasets or read-write checkpoints.

The mounting support works for object storage buckets. Supporting EBS is tracked in #2387.

**Comment by yuval-airis** [comment]
Unfortunately no. I didn't measure the required runtime, but I am thinking of somewhere along the lines of 500GB-1000GB data randomly accessible to many short runs (10s of seconds).

If I do that, I imagine I'll need to download all of that information and only then start my run which would increase my runtime significantly.

**Reply by concretevitamin** [reply]
Are you thinking of using EBS or EFS?

**Reply by yuval-airis** [reply]
@concretevitamin EFS

**Reply by concretevitamin** [reply]
Here's a quick gist demonstrating setting up EFS: https://gist.github.com/concretevitamin/eec319cea8b2bac3ac03f23f111c0561

Note the caveats at the top of the YAML. Also, EFS is region-specific (meaning the search space used by SkyPilot to launch compute is restricted to 1 region, reducing GPU availability).

If this proves useful, we could look into supporting it more natively (e.g., cross-region).
